rules/library.bzl

```diff
@@ -357,17 +358,18 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
             deps = [],
             tags = _MANUAL,
         )
-        deps.append(import_name)
+        vendored_deps.append(import_name)
     for vendored_static_library in kwargs.pop("vendored_static_libraries", []):
         import_name = "%s-%s-library-import" % (name, paths.basename(vendored_static_library))
         objc_import(
             name = import_name,
             archives = [vendored_static_library],
             tags = _MANUAL,
         )
-        deps.append(import_name)
+        vendored_deps.append(import_name)
     for vendored_dynamic_library in kwargs.pop("vendored_dynamic_libraries", []):
         fail("no import for %s" % vendored_dynamic_library)
+    deps += vendored_deps
```

Comment:
nice! thanks for keeping this a single array, it'll make reading `bazel query --output=build` output much nicer